Don't use NonVariableTypeMode in custom ops

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37355 Don't use NonVariableTypeMode in custom ops**

Potentially fixes https://github.com/pytorch/pytorch/issues/37306

Differential Revision: [D21261946](https://our.internmc.facebook.com/intern/diff/D21261946/)

smessmer committed:
Don't use NonVariableTypeMode in custom ops

Potentially fixes https://github.com/pytorch/pytorch/issues/37306

Differential Revision: [D21261946](https://our.internmc.facebook.com/intern/diff/D21261946/)

[ghstack-poisoned]

ezyang reviewed (approved):
Needs some tests, but you can do them in follow up

smessmer committed:
Update on "Don't use NonVariableTypeMode in custom ops"


Potentially fixes https://github.com/pytorch/pytorch/issues/37306

Differential Revision: [D21261946](https://our.internmc.facebook.com/intern/diff/D21261946/)

[ghstack-poisoned]

smessmer committed:
Update on "Don't use NonVariableTypeMode in custom ops"


Potentially fixes https://github.com/pytorch/pytorch/issues/37306

Differential Revision: [D21261946](https://our.internmc.facebook.com/intern/diff/D21261946/)

[ghstack-poisoned]

smessmer committed:
Update on "Don't use NonVariableTypeMode in custom ops"


Potentially fixes https://github.com/pytorch/pytorch/issues/37306

Differential Revision: [D21261946](https://our.internmc.facebook.com/intern/diff/D21261946/)

[ghstack-poisoned]
